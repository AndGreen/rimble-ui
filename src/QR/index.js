import React, { Component } from 'react';
import PropTypes from 'prop-types';
import QRCode from 'qrcode.react';

const QR = React.forwardRef((props, ref) => <QRCode ref={ref} {...props} />);

QR.propTypes = {
  /**
   * Sets the value of the QR code
   */
  value: PropTypes.string.isRequired,
};

QR.displayName = 'QR';

export default QR;
