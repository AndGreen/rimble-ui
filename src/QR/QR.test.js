import QR from './'

describe('QR component sanity', () => {
  it('has name', () => {
    expect(QR.displayName).toBe('QR')
  })

  it('matches default snapshot', () => {
    const component = render( <QR value='test' /> )
    expect(component).toMatchSnapshot()
  })
})
