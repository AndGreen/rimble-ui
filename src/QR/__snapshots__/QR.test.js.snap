// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QR component sanity matches default snapshot 1`] = `
<canvas
  height="128"
  style="height:128px;width:128px"
  width="128"
>
  
</canvas>
`;
